sympy/combinatorics/rewritingsystem.py

```diff
@@ -32,6 +33,8 @@ def __init__(self, group):
 
         # dictionary of reductions
         self.rules = {}
+        self.rules_cache = deque([])
```

Comment:
It is possible to set `maxlen` of a deque. Then its size is automatically bounded.